src/emysql.erl

```diff
@@ -809,3 +807,43 @@ monitor_work(Connection0, Timeout, Args) when is_record(Connection0, emysql_conn
         emysql_conn:reset_connection(emysql_conn_mgr:pools(), Connection, pass),
         exit(mysql_timeout)
     end.
+
+%% @spec execute_transaction(Connection, Fun) -> Result
+%%      Connection = pid()
+%%      Fun = fun()
+
+%%      Result = ok_packet() | result_packet() | error_packet()
+%%
+%% @doc Execute a transaction function
+%%
+%% @private
+%% @end doc: jfjalburquerque july 15
+%%
+execute_transaction(Connection, Fun) ->
+    Result = try
+                 case emysql_conn:execute(Connection, <<"START TRANSACTION;">>, []) of
+                     #ok_packet{} ->
+                         %% execute transaction function
+                         Fun(fun(Query, Params) -> emysql_conn:execute(Connection, Query, Params) end);
+                     MysqlError -> MysqlError
+                 end
+             catch
+                 _:Error ->
+                     %% in case of error in the transaction function
+                     %% we send #error_packet to do rollback
+                     #error_packet{msg = Error}
+             end,
+    case Result of
+        #error_packet{} ->
+            %% in case of #error_packet we execute rollback
+            case emysql_conn:execute(Connection, <<"ROLLBACK;">>, []) of
+                #ok_packet{} -> Result#error_packet{code = 1402, status = <<"XA100">>};
```

Comment:
Yes, I want to do it. I have based my code in "erlmysql" transactions, if there is a rollback it returns a mysql error (code 1402, Transaction branch was rolled back), and in the message I insert the error of the function to debug it. 

Do you think is better show the error from another way ?